// @prisma/internals/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var dist_exports = {};
__export(dist_exports, {
  BinaryType: () => import_chunk_PGE3XZUY.BinaryType,
  ClientEngineType: () => import_chunk_GLMB7J25.ClientEngineType,
  DEFAULT_CLIENT_ENGINE_TYPE: () => import_chunk_GLMB7J25.DEFAULT_CLIENT_ENGINE_TYPE,
  ErrorArea: () => import_chunk_A4QWQE3W.ErrorArea,
  Generator: () => import_chunk_2S3XA3II.Generator,
  HelpError: () => import_chunk_IVMYNCCJ.HelpError,
  RustPanic: () => import_chunk_A4QWQE3W.RustPanic,
  SchemaEngineExitCode: () => import_chunk_VBJ4ID6C.SchemaEngineExitCode,
  arg: () => import_chunk_2ZMLAV6A.arg,
  assertAlways: () => import_chunk_Y4TOGY6I.assertAlways,
  assertNever: () => import_chunk_IL7PSY44.assertNever,
  byline: () => import_chunk_L6KOIJJ5.byline,
  callOnceOnSuccess: () => import_chunk_NIWBAJZV.callOnceOnSuccess,
  canConnectToDatabase: () => import_chunk_VBJ4ID6C.canConnectToDatabase,
  canPrompt: () => import_chunk_5L2QQZK7.canPrompt,
  checkUnsupportedDataProxy: () => import_chunk_LRMWOEWW.checkUnsupportedDataProxy,
  chmodPlusX: () => import_chunk_RK3Z3Z5H.chmodPlusX,
  createDatabase: () => import_chunk_VBJ4ID6C.createDatabase,
  credentialsToUri: () => import_chunk_SWVW77BE.credentialsToUri,
  drawBox: () => import_chunk_3QVYUSJB.drawBox,
  dropDatabase: () => import_chunk_VBJ4ID6C.dropDatabase,
  engineEnvVarMap: () => import_chunk_PGE3XZUY.engineEnvVarMap,
  extractPreviewFeatures: () => import_chunk_Z7JJ2NRW.extractPreviewFeatures,
  fixBinaryTargets: () => import_chunk_5RIIWTW4.fixBinaryTargets,
  format: () => import_chunk_2ZMLAV6A.format,
  formatSchema: () => import_chunk_LRMWOEWW.formatSchema,
  formatTable: () => import_chunk_QN6PSQY7.formatTable,
  formatms: () => import_chunk_XYCIH36R.formatms,
  fsFunctional: () => import_chunk_YJYPWUHE.fs_functional_exports,
  fsUtils: () => import_chunk_SAWC6JNZ.fs_utils_exports,
  getCLIPathHash: () => import_chunk_FFBB4KHP.getCLIPathHash,
  getClientEngineType: () => import_chunk_GLMB7J25.getClientEngineType,
  getCommandWithExecutor: () => import_chunk_SDBPED3S.getCommandWithExecutor,
  getConfig: () => import_chunk_XR2ZRXE5.getConfig,
  getDMMF: () => import_chunk_GACM5RDZ.getDMMF,
  getDirectUrl: () => import_chunk_XR2ZRXE5.getDirectUrl,
  getEffectiveUrl: () => import_chunk_XR2ZRXE5.getEffectiveUrl,
  getEngineVersion: () => import_chunk_PNYSASZX.getEngineVersion,
  getEnginesMetaInfo: () => import_chunk_I2TBT2A4.getEnginesMetaInfo,
  getEnvPaths: () => import_chunk_BP2EIKDO.getEnvPaths,
  getGenerator: () => import_chunk_LRMWOEWW.getGenerator,
  getGeneratorSuccessMessage: () => import_chunk_7QF5VVOD.getGeneratorSuccessMessage,
  getGenerators: () => import_chunk_LRMWOEWW.getGenerators,
  getLintWarningsAsText: () => import_chunk_PPKYE46G.getLintWarningsAsText,
  getNodeAPIName: () => import_chunk_LRMWOEWW.getNodeAPIName,
  getPackedPackage: () => import_chunk_6HJSACPK.getPackedPackage,
  getPlatform: () => import_chunk_LRMWOEWW.getPlatform,
  getPrismaConfigFromPackageJson: () => import_chunk_BHLFQQOC.getPrismaConfigFromPackageJson,
  getProjectHash: () => import_chunk_FFBB4KHP.getProjectHash,
  getRelativeSchemaPath: () => import_chunk_BHLFQQOC.getRelativeSchemaPath,
  getSchema: () => import_chunk_BHLFQQOC.getSchema,
  getSchemaDir: () => import_chunk_BHLFQQOC.getSchemaDir,
  getSchemaPath: () => import_chunk_BHLFQQOC.getSchemaPath,
  getSchemaPathFromPackageJson: () => import_chunk_BHLFQQOC.getSchemaPathFromPackageJson,
  getSchemaPathFromPackageJsonSync: () => import_chunk_BHLFQQOC.getSchemaPathFromPackageJsonSync,
  getSchemaPathSync: () => import_chunk_BHLFQQOC.getSchemaPathSync,
  handleLibraryLoadingErrors: () => import_chunk_URNJAKMY.handleLibraryLoadingErrors,
  handleLintPanic: () => import_chunk_PPKYE46G.handleLintPanic,
  handlePanic: () => import_chunk_4LJXHCRY.handlePanic,
  hasOwnProperty: () => import_chunk_WV4QHCSM.hasOwnProperty,
  highlightDatamodel: () => import_chunk_DQ4QGRJP.highlightDatamodel,
  highlightSql: () => import_chunk_DQ4QGRJP.highlightSql,
  highlightTS: () => import_chunk_DQ4QGRJP.highlightTS,
  isCi: () => import_chunk_MSJIUJC7.isCi,
  isCurrentBinInstalledGlobally: () => import_chunk_MY5E2AKY.isCurrentBinInstalledGlobally,
  isError: () => import_chunk_2ZMLAV6A.isError,
  isInteractive: () => import_chunk_ICFLO5JW.isInteractive,
  isPromiseLike: () => import_chunk_JTKAT4KZ.isPromiseLike,
  isRustPanic: () => import_chunk_A4QWQE3W.isRustPanic,
  keyBy: () => import_chunk_ZKWDIAVF.keyBy,
  link: () => import_chunk_W4MUUWEL.link,
  lintSchema: () => import_chunk_PPKYE46G.lintSchema,
  load: () => import_chunk_MELHV4JU.loadLibrary,
  loadEnvFile: () => import_chunk_JJ4ZA37O.loadEnvFile,
  logger: () => import_chunk_NAPCALUU.logger_exports,
  longestCommonPathPrefix: () => import_chunk_J77ZIDXB.longestCommonPathPrefix,
  mapObjectValues: () => import_chunk_UGR6KLZN.mapObjectValues,
  maskSchema: () => import_chunk_GXDHVUPK.maskSchema,
  maxBy: () => import_chunk_CGFNDGGI.maxBy,
  maxWithComparator: () => import_chunk_CGFNDGGI.maxWithComparator,
  missingGeneratorMessage: () => import_chunk_BLEP247V.missingGeneratorMessage,
  parseAWSNodejsRuntimeEnvVarVersion: () => import_chunk_6UZD6PXW.parseAWSNodejsRuntimeEnvVarVersion,
  parseBinaryTargetsEnvValue: () => import_chunk_UJCWYERA.parseBinaryTargetsEnvValue,
  parseEnvValue: () => import_chunk_UJCWYERA.parseEnvValue,
  pathToPosix: () => import_chunk_J77ZIDXB.pathToPosix,
  pick: () => import_chunk_5V36RLCO.pick,
  platformRegex: () => import_chunk_46SDEADY.platformRegex,
  printConfigWarnings: () => import_chunk_JE2YWM7K.printConfigWarnings,
  printGeneratorConfig: () => import_chunk_XX47T7HE.printGeneratorConfig,
  protocolToConnectorType: () => import_chunk_SWVW77BE.protocolToConnectorType,
  resolveBinary: () => import_chunk_PGE3XZUY.resolveBinary,
  resolveUrl: () => import_chunk_XR2ZRXE5.resolveUrl,
  sendPanic: () => import_chunk_3TNHESE5.sendPanic,
  serializeQueryEngineName: () => import_chunk_HO5HRHFZ.serializeQueryEngineName,
  setClassName: () => import_chunk_ANU3HEGF.setClassName,
  trimBlocksFromSchema: () => import_chunk_YKYVNUGX.trimBlocksFromSchema,
  trimNewLine: () => import_chunk_YKYVNUGX.trimNewLine,
  tryLoadEnvs: () => import_chunk_BEYWX6OE.tryLoadEnvs,
  unknownCommand: () => import_chunk_IVMYNCCJ.unknownCommand,
  uriToCredentials: () => import_chunk_SWVW77BE.uriToCredentials,
  validate: () => import_chunk_RPRZDOO5.validate,
  vercelPkgPathRegex: () => import_chunk_M3GBKLGE.vercelPkgPathRegex,
  version: () => import_chunk_MIGBWPUV.version,
  warnOnce: () => import_chunk_6JCV7L72.warnOnce,
  wasm: () => import_chunk_CHY6NJBA.wasm_exports
});
module.exports = __toCommonJS(dist_exports);
var import_chunk_LRMWOEWW = require("./chunk-LRMWOEWW.js");
var import_chunk_EUTJYQDW = require("./chunk-EUTJYQDW.js");
var import_chunk_IPKE37QC = require("./chunk-IPKE37QC.js");
var import_chunk_AZ66WJ3B = require("./chunk-AZ66WJ3B.js");
var import_chunk_RGLQVRDT = require("./chunk-RGLQVRDT.js");
var import_chunk_R5QZ4HCI = require("./chunk-R5QZ4HCI.js");
var import_chunk_2V53YI7V = require("./chunk-2V53YI7V.js");
var import_chunk_MFF5QDB3 = require("./chunk-MFF5QDB3.js");
var import_chunk_X3P4J7WW = require("./chunk-X3P4J7WW.js");
var import_chunk_XX47T7HE = require("./chunk-XX47T7HE.js");
var import_chunk_5RIIWTW4 = require("./chunk-5RIIWTW4.js");
var import_chunk_DPKROP4G = require("./chunk-DPKROP4G.js");
var import_chunk_RS2R7COT = require("./chunk-RS2R7COT.js");
var import_chunk_4CGPN5SR = require("./chunk-4CGPN5SR.js");
var import_chunk_5V36RLCO = require("./chunk-5V36RLCO.js");
var import_chunk_46SDEADY = require("./chunk-46SDEADY.js");
var import_chunk_JE2YWM7K = require("./chunk-JE2YWM7K.js");
var import_chunk_HO5HRHFZ = require("./chunk-HO5HRHFZ.js");
var import_chunk_ANU3HEGF = require("./chunk-ANU3HEGF.js");
var import_chunk_YKYVNUGX = require("./chunk-YKYVNUGX.js");
var import_chunk_CGFNDGGI = require("./chunk-CGFNDGGI.js");
var import_chunk_HC6MILYB = require("./chunk-HC6MILYB.js");
var import_chunk_BLEP247V = require("./chunk-BLEP247V.js");
var import_chunk_6UZD6PXW = require("./chunk-6UZD6PXW.js");
var import_chunk_JTKAT4KZ = require("./chunk-JTKAT4KZ.js");
var import_chunk_ZKWDIAVF = require("./chunk-ZKWDIAVF.js");
var import_chunk_JJ4ZA37O = require("./chunk-JJ4ZA37O.js");
var import_chunk_7B5PXS6N = require("./chunk-7B5PXS6N.js");
var import_chunk_UGR6KLZN = require("./chunk-UGR6KLZN.js");
var import_chunk_4LJXHCRY = require("./chunk-4LJXHCRY.js");
var import_chunk_W4MUUWEL = require("./chunk-W4MUUWEL.js");
var import_chunk_AKGBON77 = require("./chunk-AKGBON77.js");
var import_chunk_WV4QHCSM = require("./chunk-WV4QHCSM.js");
var import_chunk_3QVYUSJB = require("./chunk-3QVYUSJB.js");
var import_chunk_Z7JJ2NRW = require("./chunk-Z7JJ2NRW.js");
var import_chunk_YJYPWUHE = require("./chunk-YJYPWUHE.js");
var import_chunk_SAWC6JNZ = require("./chunk-SAWC6JNZ.js");
var import_chunk_J77ZIDXB = require("./chunk-J77ZIDXB.js");
var import_chunk_SDBPED3S = require("./chunk-SDBPED3S.js");
var import_chunk_MY5E2AKY = require("./chunk-MY5E2AKY.js");
var import_chunk_BP2EIKDO = require("./chunk-BP2EIKDO.js");
var import_chunk_BEYWX6OE = require("./chunk-BEYWX6OE.js");
var import_chunk_HW5RCZSE = require("./chunk-HW5RCZSE.js");
var import_chunk_Y4TOGY6I = require("./chunk-Y4TOGY6I.js");
var import_chunk_IL7PSY44 = require("./chunk-IL7PSY44.js");
var import_chunk_L6KOIJJ5 = require("./chunk-L6KOIJJ5.js");
var import_chunk_NIWBAJZV = require("./chunk-NIWBAJZV.js");
var import_chunk_5L2QQZK7 = require("./chunk-5L2QQZK7.js");
var import_chunk_MSJIUJC7 = require("./chunk-MSJIUJC7.js");
var import_chunk_ICFLO5JW = require("./chunk-ICFLO5JW.js");
var import_chunk_PPKYE46G = require("./chunk-PPKYE46G.js");
var import_chunk_RPRZDOO5 = require("./chunk-RPRZDOO5.js");
var import_chunk_DQ4QGRJP = require("./chunk-DQ4QGRJP.js");
var import_chunk_OIVPX3BR = require("./chunk-OIVPX3BR.js");
var import_chunk_CYAS6B5N = require("./chunk-CYAS6B5N.js");
var import_chunk_YAD542DN = require("./chunk-YAD542DN.js");
var import_chunk_63KFPQEY = require("./chunk-63KFPQEY.js");
var import_chunk_XR2ZRXE5 = require("./chunk-XR2ZRXE5.js");
var import_chunk_GACM5RDZ = require("./chunk-GACM5RDZ.js");
var import_chunk_ZWZY7GYQ = require("./chunk-ZWZY7GYQ.js");
var import_chunk_OUUMDNA7 = require("./chunk-OUUMDNA7.js");
var import_chunk_MIGBWPUV = require("./chunk-MIGBWPUV.js");
var import_chunk_QN6PSQY7 = require("./chunk-QN6PSQY7.js");
var import_chunk_I2TBT2A4 = require("./chunk-I2TBT2A4.js");
var import_chunk_P56LTJA3 = require("./chunk-P56LTJA3.js");
var import_chunk_PNYSASZX = require("./chunk-PNYSASZX.js");
var import_chunk_MELHV4JU = require("./chunk-MELHV4JU.js");
var import_chunk_URNJAKMY = require("./chunk-URNJAKMY.js");
var import_chunk_CHY6NJBA = require("./chunk-CHY6NJBA.js");
var import_chunk_2D2BUMA4 = require("./chunk-2D2BUMA4.js");
var import_chunk_IVMYNCCJ = require("./chunk-IVMYNCCJ.js");
var import_chunk_7QF5VVOD = require("./chunk-7QF5VVOD.js");
var import_chunk_XYCIH36R = require("./chunk-XYCIH36R.js");
var import_chunk_GLMB7J25 = require("./chunk-GLMB7J25.js");
var import_chunk_FFBB4KHP = require("./chunk-FFBB4KHP.js");
var import_chunk_BHLFQQOC = require("./chunk-BHLFQQOC.js");
var import_chunk_2ZMLAV6A = require("./chunk-2ZMLAV6A.js");
var import_chunk_2B26F5N3 = require("./chunk-2B26F5N3.js");
var import_chunk_VBJ4ID6C = require("./chunk-VBJ4ID6C.js");
var import_chunk_DUMKIIDL = require("./chunk-DUMKIIDL.js");
var import_chunk_PGE3XZUY = require("./chunk-PGE3XZUY.js");
var import_chunk_M3GBKLGE = require("./chunk-M3GBKLGE.js");
var import_chunk_IFGJW36A = require("./chunk-IFGJW36A.js");
var import_chunk_RK3Z3Z5H = require("./chunk-RK3Z3Z5H.js");
var import_chunk_ABB2542Q = require("./chunk-ABB2542Q.js");
var import_chunk_3TNHESE5 = require("./chunk-3TNHESE5.js");
var import_chunk_GXDHVUPK = require("./chunk-GXDHVUPK.js");
var import_chunk_OLSA6YSL = require("./chunk-OLSA6YSL.js");
var import_chunk_4CWLFPLE = require("./chunk-4CWLFPLE.js");
var import_chunk_A4QWQE3W = require("./chunk-A4QWQE3W.js");
var import_chunk_6JCV7L72 = require("./chunk-6JCV7L72.js");
var import_chunk_2S3XA3II = require("./chunk-2S3XA3II.js");
var import_chunk_UJCWYERA = require("./chunk-UJCWYERA.js");
var import_chunk_HOZFIXMD = require("./chunk-HOZFIXMD.js");
var import_chunk_SWVW77BE = require("./chunk-SWVW77BE.js");
var import_chunk_T2PM5TKT = require("./chunk-T2PM5TKT.js");
var import_chunk_3C2LIF3Q = require("./chunk-3C2LIF3Q.js");
var import_chunk_6HJSACPK = require("./chunk-6HJSACPK.js");
var import_chunk_TCKAZCW5 = require("./chunk-TCKAZCW5.js");
var import_chunk_HM4TYOFK = require("./chunk-HM4TYOFK.js");
var import_chunk_FGKIPLS5 = require("./chunk-FGKIPLS5.js");
var import_chunk_IMQ4NNCQ = require("./chunk-IMQ4NNCQ.js");
var import_chunk_IQB47LU2 = require("./chunk-IQB47LU2.js");
var import_chunk_VWHCET2H = require("./chunk-VWHCET2H.js");
var import_chunk_W4B2MEEJ = require("./chunk-W4B2MEEJ.js");
var import_chunk_MFGQSSGY = require("./chunk-MFGQSSGY.js");
var import_chunk_6FFJQ4TS = require("./chunk-6FFJQ4TS.js");
var import_chunk_NAPCALUU = require("./chunk-NAPCALUU.js");
var import_chunk_DJGRU4RH = require("./chunk-DJGRU4RH.js");
var import_chunk_OQQ4P4PG = require("./chunk-OQQ4P4PG.js");
